PUSH_TO_FILE.md

```diff
@@ -358,6 +360,155 @@ admin-username="dev/redis/username"
+package. The following is an example of using template function to HTML
+escape/encode a secret value.
+
+```
+{{ secret "alias" | html }}
+{{ secret "alias" | urlquery }}
+```
+
+If the value retrieved from Conjur for `alias` is `"<Hello@World!>"`,
+the following file content will be rendered, each HTML escaped and encoded,
+respectively:
+
+```
+&lt;Hello;@World!&gt;
+%3CHello%40World%21%3E
+```
+
+For a full list of global Go text template functions, reference the official
+[`text/template` documentation](https://pkg.go.dev/text/template#hdr-Functions).
+
+### Execution "Double-Pass"
+
+To avoid leaking sensitive secret data to logs, and to ensure that a
+misconfigured Push to File workflow fails fast, Push to File implements a
+"double-pass" execution of custom templates. The template "first-pass" runs
+before secrets are retrieved from Conjur, and validates that the provided custom
+template successfully executes given `"REDACTED"` for each secret value.
+Redacting secret values allows for secure, complete error logging for
+malformed templates. The template "second-pass" runs when rendering secret
+files, and error messages during this stage are sanitized. Custom templates that
+pass the "first-pass" and fail the "second-pass" require experimenting locally
+to identify bugs.
+
+_**NOTE**: Custom templates should not branch conditionally on secret values.
+This may result in a template first-pass execution that doesn't validate all
+branches of the custom template._
+
+### Example Custom Templates: Direct reference to secret values
+
+The following is an example of using a custom template to render secret data by
+referencing secrets directly using the custom template function `secret`.
+
+```
+conjur.org/secret-file-path.direct-reference: ./direct.txt
+conjur.org/secret-file-template.direct-reference: |
+  username | {{ secret "db-username" }}
+  password | {{ secret "db-password" }}
+```
+
+Assuming that the following secrets have been retrieved for secret group
+`direct-reference`:
+
+```
+db-username: admin
+db-password: my$ecretP@ss!
+```
+
+Secrets Provider will render the following content for the file
+`/conjur/secrets/direct.txt`:
+
+```
+username | admin
+password | my$ecretP@ss!
+```
+
+### Example Custom Templates: Iterative approach
+
+The following is an example of using a custom template to render secret data
+using an iterative process instead of referencing all variables directly.
+
+```
+conjur.org/secret-file-path.iterative-reference: ./iterative.txt
+conjur.org/secret-file-template.iterative-reference: |
+  {{- range $index, $secret := .SecretsArray -}}
+  {{- $secret.Alias }} | {{ $secret.Value }}
+  {{- end -}}
+```
+
+Here, `.SecretsArray` is a reference to Secret Provider's internal array of
+secrets that have been retrieved from Conjur. For each entry in this array,
+there is a secret `Alias` and `Value` field tat can be referenced in the custom
```

Comment:
If you have to rebase anyway, there's a typo: `tat can be referenced` should be `that can be referenced`.

Reply:
Rats, my bad. Fixed, rebased, and (officially) opened for review.